abi/encoding_test.go

```diff
@@ -760,3 +761,71 @@ func TestEncodingStruct_camcelCase(t *testing.T) {
 		t.Fatal("bad")
 	}
 }
+
+func TestEncodingStruct_dynamicTuple(t *testing.T) {
+	typ := MustNewType("tuple(address aa, uint32[] b)")
+
+	type Obj struct {
+		A ethgo.Address `abi:"aa"`
+		B []uint32
+	}
+	obj := Obj{
+		A: ethgo.Address{0x1},
+		B: []uint32{1, 2},
+	}
+
+	encoded, err := typ.Encode(&obj)
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
FYI, you can use this as a short-hand.

```suggestion
	require.NoError(t, err)
```